runtime/src/main/java/org/corfudb/util/serializer/JsonSerializer.java

```diff
@@ -58,7 +58,7 @@ public Object deserialize(ByteBuf b, CorfuRuntime rt) {
             try {
                 return rt.getObjectsView().build()
                         .setStreamID(new UUID(b.readLong(), b.readLong()))
-                        .setType(Class.forName(smrClassName))
+                        .setType((Class<? extends ISMRObject>) Class.forName(smrClassName))
```

Comment:
Yea, these casts can be cleaned up once we directly use ISMRObject.

Reply:
Looks like they are still needed. The current implementation is:

`<R extends ICorfuSMR<R>> IObjectBuilder<R> setType(Class<R> type);`